Parsers/ASimWebSession/Parsers/ASimWebSessionPaloAltoNetworks.yaml

```diff
@@ -0,0 +1,98 @@
+Parser:
+  Title: Web Session ASIM parser for Palo Alto Networks URL Filtering
+  Version: '0.1'
+  LastUpdated: Jan 25 2023
+Product:
+  Name: Palo Alto Networks
+Normalization:
+  Schema: WebSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: URL Filter fields
+  Link: https://docs.paloaltonetworks.com/pan-os/11-0/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/url-filtering-log-fields.html
+- Title: Palo Alto Common Event Format Integration Guide [pdf]
+  Link: https://docs.paloaltonetworks.com/content/dam/techdocs/en_US/pdf/cef/pan-os-10-0-cef-configuration-guide.pdf
+Description: |
+  This ASIM parser supports normalizing the Palo Alto Networks CEF log to the ASIM WebSession normalized schema.
+ParserName: ASimWebSessionPaloAltoNetworks
+EquivalentBuiltInParser: _ASim_WebSession_PaloAltoNetworks
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser=(disabled:bool=false){
+      let EventLookup=datatable(DeviceAction:string,EventResult:string,HttpStatusCode:string)
+      [
+          "alert", "Success","200"
+          , "allow", "Success","200"
+          , "continue", "Success","200"
+          , "override", "Success","200"
+          , "block-continue", "Partial","200"
+          , "block-url", "Failure","503"
+          , "block-override", "Failure","302"
+          , "override-lockout", "Failure","503"
+      ];
+      let SeverityLookup=datatable(LogSeverity:string,EventSeverity:string)
+      [   1, "Informational" 
+          , 2, "Low" 
+          , 3, "Medium"
+          , 4, "Medium" 
+          , 5, "High"
+      ];
+      CommonSecurityLog
+      | where DeviceVendor == "Palo Alto Networks"
+          and DeviceProduct == "PAN-OS"
+          and Activity == "THREAT"
+          and DeviceEventClassID == "url"
+      | parse AdditionalExtensions with URL ";PanOSActionFlags=" *
```

Comment:
1. I did not find in the sample data any AdditionalExtension that had a URL before PanOSActionFlags.
2. I would parse with the preceding key name for the URL, rather than the following key name as order can change.
3. lastly, AdditionlExtensions have additional values we need to extaract (using parse-kv ofcourse). Some I identified are:
   - PanOSXForwarderfor (or PanXFFIP, I wander which)
   - PanOSReferer
   - PanOSThreatCategory
   - PanOSRuleUUID
   - PanOSURLCatList
   - PanDstHostname, PanDstMac.
   - PanSrcHostname, PanSrcMac

Other Dst/Src fields can also be extracted while we are at it, even though they don't have direct corresponding ASIM fields (SrcDvcOs for example)

Reply:
All can have data. It depends on the firewall setup. Let's map them.

Reply:
Mapped all src and dst fields